core/provider/service.go

```diff
@@ -260,24 +260,31 @@ func (s *Service) FetchResources(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-
 	failedProviders := map[string]error{}
+	resourcesCount := 0
+	upsertedResourceCount := 0
 	for _, p := range providers {
 		startTime := time.Now()
 		s.logger.Info(ctx, "fetching resources", "provider_urn", p.URN)
-		resources, err := s.getResources(ctx, p)
+		newResourcesCount, resources, err := s.fetchNewResources(ctx, p)
```

Comment:
let's change `newResourcesCount` to `fetchedResourcesCount` or similar

Reply:
and would be better if the primary information is returned in the first order -> `resources, fetchedResourcesCount, err := ...`